src/app/pages/courses/pipes/orderby.pipe.ts

```diff
@@ -0,0 +1,22 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { Course } from 'app/shared/models/course.model';
+
+@Pipe({
+    name: 'orderByDate'
+})
+
+export class OrderByPipe implements PipeTransform {
+    public transform(courses: Course[], direction: string) {
+        if (direction === 'asc') {
+            return courses.sort((a: Course, b: Course) => {
```

Comment:
Native `sort` function is mutative. It modifies the array it's being applied to. This is bad for your application, because it makes this pipe to be impure. Try using something like `lodash/sortBy` library to achieve the result.